src/api/controller/carpool/carpool_controller.py

```diff
@@ -0,0 +1,99 @@
+from database.exceptions import InvalidPrice
+
+carpool_blueprint = Blueprint("carpool", __name__, url_prefix="/carpool")
+
+def is_valid_hour(hour):
+    try:
+        datetime.fromtimestamp(float(hour))
+        return True
+    except ValueError:
+        return False
+
+
+def is_valid_location(location):
+    query = "INSERT INTO carmate.location_table (location) VALUES (%s) RETURNING *"
+    conn = None
+    try:
+        conn = connect(user="your_user", password="your_password", host="your_host", database="your_database")
+        with conn.cursor() as curs:
+            curs.execute(query, (location,))
+            result = curs.fetchone()
+    except UniqueViolation as e:
+        raise e
+    except Exception as e:
+        raise e
+    finally:
+        if conn:
+            conn.close()
+    return True
+
+
+def is_valid_price(price: float) -> None:
+    if price < 0:
+        raise InvalidPrice("Price cannot be negative.")
+
+    price_str = str(price)
+    decimal_index = price_str.find(".")
+    
+    if decimal_index != -1 and len(price_str[decimal_index + 1:]) > 2:
+        raise InvalidPrice("Price cannot have more than two digits after the decimal point.")
+    
+def is_valid_email(email):
+    query = "SELECT * FROM carmate.email_table WHERE email = %s"
+    conn = None
+    try:
+        conn = establishing_connection()
+        with conn.cursor() as curs:
+            curs.execute(query, (email,))
+            result = curs.fetchone()
+    except Exception as e:
+        raise InternalServer(str(e))
+    finally:
+        if conn:
+            conn.close()
+    if result:
+        raise UniqueViolation("Email already exists")
+    return True
+
+
+@carpool_blueprint.route("/create", methods=["POST"])
+def create_carpool():
+    try:
+        start_hour = request.json['start_hour']
+        localisation_A = request.json['localisation_A']
+        localisation_B = request.json['localisation_B']
+        price = request.json['price']
+        email = request.json['email']
+
+        if not start_hour or not localisation_A or not localisation_B or not price or not email:
+            return jsonify({'error': 'All fields are required'}), 400
+
+        if not is_valid_email(email):
+            return jsonify({'error': 'Invalid email format'}), 400
+        
+        if not is_valid_hour(start_hour):
+            return jsonify({'error': 'Invalid hour format'}), 400
+
+        if not is_valid_location(localisation_A) or not is_valid_location(localisation_B):
+            return jsonify({'error': 'Invalid location format'}), 400
+
+        carpool_dto = CarpoolDTO(
+            start_hour=start_hour,
+            localisation_A=localisation_A,
+            localisation_B=localisation_B,
+            price=price,
+            email=email
+        )
+
+        result = carpool_dto.create_carpool(carpool_dto)
+        return jsonify({'message': 'Carpool created successfully', 'carpool_id': result.id}), 204
```

Comment:
```suggestion
        return '', 204
```
`204` veut dire qu'il y a pas de contenu dans la requête
Lire: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

Soit tu renvoie un json qui possède le `carpool_id` avec une `200` soit une `204` avec rien du tout